### PR TITLE
feat: input generics

### PR DESCRIPTION
**Pull request by geekysilento**

### CODE REVIEW

**Comment by geekysilento** [comment]
no more crlf BS, only clean source code.

**Comment by GetPsyched** [comment]
Awesome

**Review by GetPsyched — CHANGES_REQUESTED** [review]
There's so much to review :(

Please finish this off from your end along with polishing and then I can review and merge quickly

File: components.json
[the remark this answers is not in the document]
Comment (geekysilento) [reply components.json:1]: this is a file used to initialize shadcn, in here u can modify things like the default theme shadcn uses, where its components will be installed to, etc

**Comment by geekysilento** [comment]
@GetPsyched how should i handle Single and Multi Select (with custom data set in drop down)? How will we populate the dropdown list? will it be defined when calling the element like <DropDown items = {itemList} /> ? and the array itemList will be handled by the form builder?

also confused about date-time plus its usecare? i have a date input generic and a time input generic, why do we need a separate single generic?

Third thing, fetching countryCodes for the telephone input generic, I still think its overkill fetching from a external source(currently idk where to fetch it from, as the site you used doesnt expose JUST the dialing codes in a specific route), my current implementation of an array of dialing codes, imo is fine. Opinions?

**Comment by GetPsyched** [comment]
> also confused about date-time plus its usecare? i have a date input generic and a time input generic, why do we need a separate single generic?

Some inputs would be only dates and some only times. For example, an all-day event would just have date, a lecture would just have time, but an exam would have both.

> Third thing, fetching countryCodes for the telephone input generic, I still think its overkill fetching from a external source(currently idk where to fetch it from, as the site you used doesnt expose JUST the dialing codes in a specific route), my current implementation of an array of dialing codes, imo is fine. Opinions?

It is fine, let's leave this for later.

**Comment by GetPsyched** [comment]
Can you give a summary of what has changed?

**Comment by GetPsyched** [comment]
Closed in favour of #102

**Pull request closed without merging**